src/Microsoft.AspNet.Owin/OwinFeatureCollection.cs

```diff
@@ -155,6 +155,16 @@ void IHttpResponseFeature.OnSendingHeaders(Action<object> callback, object state
             register(callback, state);
         }
 
+        void IHttpResponseFeature.OnResponseCompleted(Action<object> callback, object state)
```

Comment:
Nah, the code you pointed supports calling the `server.OnSendingHeaders` callback from ASP.NET 5, but can't inject `server.OnSendingHeaders` in the OWIN environment, which is what the `OwinEnvironment` class does.

Anyway, @Tratcher said they were only using community defined keys.